Src/Newtonsoft.Json/Serialization/JsonSerializerInternalReader.cs

```diff
@@ -169,9 +169,12 @@ public object Deserialize(JsonReader reader, Type objectType, bool checkAddition
 
                 if (checkAdditionalContent)
                 {
-                    if (reader.Read() && reader.TokenType != JsonToken.Comment)
+                    while (reader.Read())
                     {
-                        throw new JsonSerializationException("Additional text found in JSON string after finishing deserializing object.");
+                        if (reader.TokenType != JsonToken.Comment)
+                        {
+                            throw new JsonSerializationException("Additional text found in JSON string after finishing deserializing object.");
```

Comment:
It's a tricky balance. There have certainly been times over the last month or so when I wished Json.NET was less informative in its error messages, as it would allow for some internal simplification, but there have certainly also been times when the line/column info has been useful to me as a user.
In this case there seems to be little to gain in not having the more informative message. The reader has to be there anyway, and once it's there the line/column info can be extracted.
I think I'll include that with the work I'm currently on that catches the relatives of this issue.